tests/files/complicated.yaml

```diff
@@ -9,3 +9,16 @@ a:
     reference: ${a.complicated}
   complicated.reference: value with ${a.complicated.2019.a} reference in it
   '2019-03-28': 2019-03-28
+
+different.types:
+  - a string
+  - true
+  - 42.0
+  - also: a mapping
+    containing:
+      multiple: keys, just to make it more complicated
+      surprise:
+        - another
+        - sequence_with: a mapping inside it, with ${a.complicated.2019.a} reference (mind = blown)
```

Comment:
Both YAML and Python will accept this, but there was a discussion previous (might have been IRL, can't remember) about this. The challenge is the way we're using dotted keys and splitting those and the like. To avoid complicating the way these values can be retrieved, `confidence` just refuses non-`str` keys at the moment. 

We could change this at some point, but it would have to be made smooth, clear and deterministic. I'm thinking that combination might be a bit a challenge :)